src/pages/task-detail-page/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import { RootState } from "../../store/store";
+import { useParams } from "react-router";
+import TaskDetail from "../../components/task-detail/TaskDetail";
+import { ToDoItem } from "../../interfaces/interfaces";
+
+type NumberParams = {
+  id: string;
+};
+
+function TaskDetailPage() {
+  const list = useSelector((state: RootState) => state.list.list);
+  const { id } = useParams<NumberParams>();
+  const idNumber = parseInt(id);
+  const selectedItem = list.filter((item: ToDoItem) => item.id === idNumber);
```

Comment:
Instead of making `selectedItem: ToDoItem[]`, you could do 
`const selectedItem = list.filter((item: ToDoItem) => item.id === idNumber)[0]` to grab the first one instead of making it an array. You may have to do some checks to make sure the array isn't empty.